core/src/test/java/io/confluent/rest/ApiHeadersTest.java

```diff
@@ -0,0 +1,166 @@
+import org.apache.http.ssl.SSLContexts;
+import org.apache.kafka.common.config.types.Password;
+import org.apache.kafka.test.TestSslUtils;
+import org.apache.kafka.test.TestSslUtils.CertificateBuilder;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class ApiHeadersTest {
+
+
+  private static String httpUri = "http://localhost:8080";
+  private static String httpsUri = "https://localhost:8081";
+
+  private static final String SSL_PASSWORD = "test1234";
+
+  private static String clientKeystoreLocation;
+  private static TestApplication app;
+
+  @BeforeClass
+  public static void setUp() throws Exception {
+    File trustStore = File.createTempFile("ApiHeadersTest-truststore", ".jks");
+    File clientKeystore = File.createTempFile("ApiHeadersTest-client-keystore", ".jks");
+    File serverKeystore = File.createTempFile("ApiHeadersTest-server-keystore", ".jks");
+
+    clientKeystoreLocation = clientKeystore.getAbsolutePath();
+
+    Map<String, X509Certificate> certs = new HashMap<>();
+    createKeystoreWithCert(clientKeystore, "client", certs);
+    createKeystoreWithCert(serverKeystore, "server", certs);
+    TestSslUtils.createTrustStore(trustStore.getAbsolutePath(), new Password(SSL_PASSWORD), certs);
+
+    Properties props = new Properties();
+    props.put(RestConfig.LISTENERS_CONFIG, httpUri + "," + httpsUri);
+    props.put(RestConfig.SSL_KEYSTORE_LOCATION_CONFIG, serverKeystore.getAbsolutePath());
+    props.put(RestConfig.SSL_KEYSTORE_PASSWORD_CONFIG, SSL_PASSWORD);
+    props.put(RestConfig.SSL_KEY_PASSWORD_CONFIG, SSL_PASSWORD);
+
+    app = new TestApplication(new TestRestConfig(props));
+    app.start();
+  }
+
+  @AfterClass
+  public static void teardown() throws Exception {
+    if (app != null) {
+      app.stop();
+    }
+  }
+
+  private static void createKeystoreWithCert(File file, String alias, Map<String, X509Certificate> certs) throws Exception {
+    KeyPair keypair = TestSslUtils.generateKeyPair("RSA");
+    CertificateBuilder certificateBuilder = new CertificateBuilder(30, "SHA1withRSA");
+    X509Certificate cCert = certificateBuilder.sanDnsName("localhost")
+        .generate("CN=mymachine.local, O=A client", keypair);
+    TestSslUtils.createKeyStore(file.getPath(), new Password(SSL_PASSWORD), alias, keypair.getPrivate(), cCert);
+    certs.put(alias, cCert);
+  }
+
+  @Test
+  public void testHttpDoesNotReturnJettyServerVersionHeader() throws Exception {
+
+    HttpGet httpget = new HttpGet(httpUri +  "/test/endpoint");
+
+    try ( CloseableHttpClient httpclient = HttpClients.createDefault();
+        CloseableHttpResponse response = httpclient.execute(httpget) ) {
+
+      int statusCode = response.getStatusLine().getStatusCode();
+      Assert.assertEquals(200, statusCode);
+
+      Header jettyServerVersionHeader = response.getFirstHeader( "Server" );
+      assertNull(jettyServerVersionHeader);
```

Comment:
nit: use Hamcrest's `assertThat` for all new code.

```suggestion
      assertThat(jettyServerVersionHeader, is(nullValue()));
```

Reply:
done